apps/ecommerce/lambda/src/controllers/apiController.ts

```diff
@@ -21,20 +15,34 @@ const ApiController = {
   ) => {
     try {
       const resourceLink: ExternalResourceLink = req.body;
-      const resourceProvider = resourceLink.sys.linkType?.split(':')[0];
-      const proxyUrl = PROVIDERS[resourceProvider?.toLowerCase()];
+      const resourceProvider = req.header('x-contentful-data-provider');
+      const proxyBaseUrl = req.header('x-contentful-data-provider-baseurl');
 
-      if (!proxyUrl) {
+      if (!proxyBaseUrl) {
         return res.status(404).send({
           status: 'error',
-          message: `Provider${resourceProvider ? `: ${resourceProvider}` : ''} not found`,
+          message: `Base URL configuration for provider${
+            resourceProvider ? `: "${resourceProvider}"` : ''
+          } not found`,
         });
       }
 
+      // TODO: fetch provider config from `${proxyBaseUrl}/config.json`
+      const proxyResourceUrl = new URL(proxyBaseUrl);
+      proxyResourceUrl.pathname += `/resource`;
+
+      // prevent SSRF exploits
+      // TODO: refactor this to get the proxy resource url from the app definition
+      // configuration parameters once this is an internal Contentful service
+      if (proxyResourceUrl.hostname !== config.domainName) {
+        // temporary: only support data provider proxying on the same domain to avoid SSRF exploits
+        throw new Error(`Invalid data provider base URL. Hostname must be ${config.domainName}`);
+      }
```

Comment:
@jsdalton PTAL at the temporary SSRF exploit remediation strategy. Open to better ideas, but I think we have to make this an internal Contentful service